cloud/scope/powervs_machine.go

```diff
@@ -424,6 +424,12 @@ func (m *PowerVSMachineScope) createIgnitionData(data []byte) (string, error) {
 	}
 
 	objHost := fmt.Sprintf("%s.s3.%s.%s", bucket, region, cosURLDomain)
+
+	cosServiceEndpoint := endpoints.FetchEndpoints(string(endpoints.COS), m.ServiceEndpoint)
+	if cosServiceEndpoint != "" {
+		m.Logger.V(3).Info("Overriding the default COS endpoint in bootstrap ignition", "cosEndpoint", cosServiceEndpoint)
```

Comment:
In general it may not be only for bootstrap ignition URL, How about this
```suggestion
		m.Logger.V(3).Info("Overriding the default COS endpoint in ignition URL", "cosEndpoint", cosServiceEndpoint)
```

Reply:
Done!